code/edgeconsent/src/main/java/com/adobe/marketing/mobile/edge/consent/ConsentExtension.java

```diff
@@ -118,40 +60,68 @@ protected String getVersion() {
 	}
 
 	/**
-	 * Call this method with the EventHub's Boot event to handle the boot operation of the {@code Consent} Extension.
 	 * <p>
-	 * On boot share the initial consents loaded from persistence to XDM shared state.
-	 *
-	 * @param event the boot {@link Event}
+	 * Called during the Consent extension's registration.
+	 * The ConsentExtension listens for the following {@link Event}s:
+	 * <ul>
+	 *     <li> {@code EventType#EDGE} and EventSource {@Code EventSource#CONSENT_PREFERENCE}</li>
+	 *     <li> {@code EventType#CONSENT} and EventSource {@Code EventSource#UPDATE_CONSENT}</li>
+	 *     <li> {@Code EventType#CONSENT} and EventSource {@Code EventSource#REQUEST_CONTENT}</li>
+	 *     <li> {@Code EventType#CONFIGURATION} and EventSource {{@Code EventSource#RESPONSE_CONTENT}</li>
+	 * </ul>
+	 * <p>
+	 */
+	@Override
+	protected void onRegistered() {
+		getApi()
+			.registerEventListener(
+				EventType.EDGE,
+				EventSource.CONSENT_PREFERENCE,
+				this::handleEdgeConsentPreferenceHandle
+			);
+		getApi().registerEventListener(EventType.CONSENT, EventSource.UPDATE_CONSENT, this::handleConsentUpdate);
+		getApi().registerEventListener(EventType.CONSENT, EventSource.REQUEST_CONTENT, this::handleRequestContent);
+		getApi()
+			.registerEventListener(
+				EventType.CONFIGURATION,
+				EventSource.RESPONSE_CONTENT,
+				this::handleConfigurationResponse
+			);
+
+		handleInitialization();
+	}
+
+	/**
+	 * Share the initial consents loaded from persistence to XDM shared state.
 	 */
-	void handleEventHubBoot(final Event event) {
-		// share the initial XDMSharedState on bootUp
+	void handleInitialization() {
+		// share the initial XDMSharedState onRegistered
 		final Consents currentConsents = consentManager.getCurrentConsents();
 
 		if (!currentConsents.isEmpty()) {
-			shareCurrentConsents(event);
+			shareCurrentConsents(null);
```

Comment:
Can you update the `shareCurrentConsents` method comment to state the `event` parameter may be null on the first call when the extension initializes?